src/pudl_archiver/depositors/zenodo.py

```diff
@@ -0,0 +1,319 @@
+
+        First creates a new version, which is a snapshot of the old one, then
+        updates that version with a new version number.
+
+        Args:
+            deposition: the deposition you want to get the new version of.
+
+        Returns:
+            A new Deposition that is a snapshot of the old one you passed in,
+            with a new major version number.
+        """
+        if not deposition.submitted:
+            return deposition
+
+        url = f"{self.api_root}/deposit/depositions/{deposition.id_}/actions/newversion"
+
+        # Create the new version
+        # When the API creates a new version, it does not return the new one.
+        # It returns the old one with a link to the new one.
+        response = await self.request(
+            "POST", url, log_label="Creating new version", headers=self.auth_write
+        )
+        old_deposition = Deposition(**response)
+
+        source_metadata = old_deposition.metadata.dict(by_alias=True)
+        metadata = {}
+        for key, val in source_metadata.items():
+            if key not in ["doi", "prereserve_doi", "publication_date"]:
+                metadata[key] = val
+
+        previous = semantic_version.Version(source_metadata["version"])
+        version_info = previous.next_major()
+
+        metadata["version"] = str(version_info)
+
+        # Update metadata of new deposition with new version info
+        data = json.dumps({"metadata": metadata})
+
+        # Get url to newest deposition
+        new_deposition_url = old_deposition.links.latest_draft
+        headers = {
+            "Content-Type": "application/json",
+        } | self.auth_write
+
+        response = await self.request(
+            "PUT",
+            new_deposition_url,
+            log_label=f"Updating version number from {previous} ({old_deposition.id_}) to {version_info}",
+            data=data,
+            headers=headers,
+        )
+        return Deposition(**response)
+
+    async def publish_deposition(self, deposition: Deposition) -> Deposition:
+        """Publish a deposition.
+
+        Needs to have at least one file, and needs to not already be published.
+
+        Args:
+            deposition: the deposition you want to publish.
+        """
+        url = deposition.links.publish
+        headers = {
+            "Content-Type": "application/json",
+        } | self.auth_actions
+        response = await self.request(
+            "POST", url, log_label="Publishing deposition", headers=headers
+        )
+        return Deposition(**response)
+
+    async def delete_file(self, deposition: Deposition, target: str) -> None:
+        """Delete a file from a deposition.
+
+        Args:
+            deposition: the deposition you are applying this change to.
+            target: the filename of the file you want to delete.
+
+        Returns:
+            None if success.
+        """
+        candidates = [f for f in deposition.files if f.filename == target]
```

Comment:
Is it possible to have more than one match here?

Reply:
I don't expect one, but I'm wary of Funky Zenodo State so I decided to make sure.